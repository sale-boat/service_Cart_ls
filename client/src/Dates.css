.bold{
    font-weight : bold;
}

.expectedDates{
    margin-top : 5px;
}
.oneDay{
    margin-top: 15px;
}

.oneDayShipping{
    color :  #555;
    font-weight : bold;
}

.stock{
    margin-top : 15px;
}

.available{
    color : #008a00;
    font-size : 17px;
}

.location{
    font-size : 13px;
}