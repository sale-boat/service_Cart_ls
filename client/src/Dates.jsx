import React from 'react';
import dates from './Dates.css';

const Dates = () => {
  const date = new Date();
  const dayNumber = date.getDate();
  const weekday = date.getDay();
  const month = date.getMonth();
  const daysOfWeek = {
    0: 'Sun',
    1: 'Mon',
    2: 'Tues',
    3: 'Wed',
    4: 'Thurs',
    5: 'Fri',
    6: 'Sat',
    7: 'Sun',
    8: 'Mon',
    9: 'Tues',
    10: 'Wed',
    11: 'Thurs',
    12: 'Fri',
    13: 'Sat',

  };
  const months = {
    0: 'Jan',
    1: 'Feb',
    2: 'Mar',
    3: 'Apr',
    4: 'May',
    5: 'Jun',
    6: 'Jul',
    7: 'Aug',
    8: 'Sep',
    9: 'Oct',
    10: 'Nov',
    11: 'Dec',
  };
  const actualMonth = months[month];
  const actualWeekday = daysOfWeek[weekday];
  const fourDaysLater = daysOfWeek[weekday + 4];
  const hour = date.getHours();
  const minutes = date.getMinutes();
  return (
    <div className={dates.expectedDates}>
      <div className={dates.free}>
        <span className={dates.bold}>
          {` Get it ${actualWeekday}, ${actualMonth} ${dayNumber} - ${fourDaysLater}, ${actualMonth} ${dayNumber + 4} `}
        </span>
        {' if you choose '}
        <span className={dates.bold}> 
          {'FREE Shipping'}
        </span>
        {' at checkout. '}
      </div>

      <div className={dates.oneDay}>
        <span className={dates.bold}>
          {` Get it Tomorrow, ${actualMonth} ${dayNumber + 1}`}
        </span>
        <span className={dates.sameDay}>
          {` if you order within ${24 - hour} hrs ${60 - minutes} mins and choose paid `}
        </span>
        <span className={dates.oneDayShipping}> One-Day Shipping</span>
        <span className={dates.sameDay}> at checkout </span>
      </div>
      <div className={dates.stock}>
        <span className={dates.available}>
          {' In Stock. '} 
          <br />
        </span>
        <span className={dates.location}> 
          {' Ships from and sold by Amazon.com. '}
        </span>
      </div>
    </div>
  );
};

export default Dates;
