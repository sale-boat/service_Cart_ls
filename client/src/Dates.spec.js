import React from 'react';
import { shallow } from 'enzyme';
import Dates from './Dates.jsx';

describe('Dates Component', () => {
  it('should render', () => {
    const dates = shallow(<Dates />);
    expect(dates).toMatchSnapshot();
  });
});