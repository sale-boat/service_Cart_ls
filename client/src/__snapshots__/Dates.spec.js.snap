// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dates Component should render 1`] = `
<div
  className="expectedDates"
>
  <div
    className="free"
  >
    <span
      className="bold"
    >
       Get it Fri, Feb 15 - Tues, Feb 19 
    </span>
     if you choose 
    <span
      className="bold"
    >
      FREE Shipping
    </span>
     at checkout. 
  </div>
  <div
    className="oneDay"
  >
    <span
      className="bold"
    >
       Get it Tomorrow, Feb 16
    </span>
    <span
      className="sameDay"
    >
       if you order within 1 hrs 12 mins and choose paid 
    </span>
    <span
      className="oneDayShipping"
    >
       One-Day Shipping
    </span>
    <span
      className="sameDay"
    >
       at checkout 
    </span>
  </div>
  <div
    className="stock"
  >
    <span
      className="available"
    >
       In Stock. 
      <br />
    </span>
    <span
      className="location"
    >
       Ships from and sold by Amazon.com. 
    </span>
  </div>
</div>
`;
